src/lava/lib/dl/netx/utils.py

```diff
@@ -0,0 +1,173 @@
+# Copyright (C) 2022 Intel Corporation
+# SPDX-License-Identifier:  BSD-3-Clause
+
+"""HDF5 net description manipulation utilities."""
+
+from typing import Tuple, Union
+import h5py
+import numpy as np
+from enum import IntEnum, unique
+
+
+@unique
+class SYNAPSE_SIGN_MODE(IntEnum):
+    """Enum for synapse sign mode. Options are {``MIXED : 1``,
+    ``EXCITATORY : 2`` and ``INHIBITORY : 2``}.
+    """
+    MIXED = 1
+    EXCITATORY = 2
+    INHIBITORY = 3
+
+
+class NetDict:
+    """Provides dictionary like access to h5py object without the h5py quirks
+
+    Parameters
+    ----------
+    filename : str or None, optional
+        filename of h5py file to be loaded. It is only invoked if hdf5 file
+        handle ``f`` is ``None``. Default is None.
+    mode : str, optional
+        file open mode, by default 'r'.
+    f : h5py.File or h5py.Group, optional
+        hdf5 file object handle. Overwrites the function of filename if it is
+        not ``None``. Default is None.
+    """
+    def __init__(
+        self,
+        filename: Union[str, None] = None,
+        mode: str = 'r',
+        f: Union[h5py.File, h5py.Group] = None,
+    ) -> None:
+        self.f = h5py.File(filename, mode) if f is None else f
+        self.str_keys = ['type']
+        self.array_keys = [
+            'shape', 'stride', 'padding', 'dilation', 'groups', 'delay',
+            'iDecay', 'refDelay', 'scaleRho', 'tauRho', 'theta', 'vDecay',
+            'vThMant', 'wgtExp',
+        ]
```

Comment:
Will maintaining this list of keys be problematic? I would expect these might change/expand regularly. Is there a way to extract them from the file, or scan a list of classes in a default ctor arg for attributes?

Reply:
Yeah this would grow as we support more features and neurons through hdf5 config. Perhaps we can find a better way of managing it in future.